### PR TITLE
Bug Fixes.

### PR DESCRIPTION
**Pull request by nathanmilot**
- Fix question hiding under send button. Was fixed, but needed to redo after other changes to improve the ui broke it, should be solved now.

  - Fix bug where users can sign up with the username Anon or some caps/lowercase form. Previously only checking for exact string match.

### CODE REVIEW

File: src/main/java/org/query/handlers/UserHandlers.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ public Long createUser(@RequestBody User user)
     throws BadRequestException, InternalException {
     logger.info("Received createUser request with username: {}", user.getUname());
 
-    if(!user.getEmail().contains("@") || user.getPassword().length() < 3 || user.getUname().equals("anon") || user.getUname().equals("Anonymous"))
+    if(!user.getEmail().contains("@") || user.getPassword().length() < 3 || user.getUname().toLowerCase().equals("anon") || user.getUname().toLowerCase().equals("anonymous"))
```
Comment (kylejustbenson) [inline src/main/java/org/query/handlers/UserHandlers.java]: I wasn't aware we were doing this I'm not sure we should. We don't prevent profanity or other pseudo-anonymous tags like "69playboy420", so I don't know why we would want to prevent these.

If we are trying to remove the anonymity of names, we should implement our email verification we have discussed.

Comment (nathanmilot) [reply src/main/java/org/query/handlers/UserHandlers.java]: The goal of this isn't to remove anonymity. It essentially just reserves the usernames anon and anonymous (or some variation of them) so that when Anonymous is displayed as a user name, it is because they are logged in as guest and not because some user set that as their name. This will matter more when/if we get to setting org/boar level anonymous controls (like allowing or disallowing anonymous users). 

anon is blocked because it is used as a check at some point to determine if a user is anon to skip some auth checks.

TL;DR
- Didn't want to confuse a prof/admin if they disable anonymous users and someone has the username anonymous
- anon is a reserved username to bypass some login checks (password matching I believe) for real anonymous users

Comment (kylejustbenson) [reply src/main/java/org/query/handlers/UserHandlers.java]: That makes sense I guess, but I'm not sure why we would prevent a user from choosing Anonymous as a username. It just seems like two separate concerns. Maybe we could have a different way of representing an anonymous user than the displayed username on a question? I don't think it's worth the effort.

Comment (nathanmilot) [reply src/main/java/org/query/handlers/UserHandlers.java]: True. We could use an anonymous logo or something?

Comment (nathanmilot) [reply src/main/java/org/query/handlers/UserHandlers.java]: like this: https://fontawesome.com/icons/user-secret?style=solid

Comment (kylejustbenson) [reply src/main/java/org/query/handlers/UserHandlers.java]: Yeah that would be really cool.

Comment (nathanmilot) [reply src/main/java/org/query/handlers/UserHandlers.java]: ok, do we want to make that another card and use these changes?

Comment (kylejustbenson) [reply src/main/java/org/query/handlers/UserHandlers.java]: Sounds good.
